### ISSUE TITLE
Bug: Incorrect parsing of apostrophes

### ISSUE THREAD

**Issue by MartinBoge** [EmilStenstrom/django-components#752]
Hi Emil

First off - thank you very much for this sweet sweet library. If this issue is redundant or you find it irrelevant just go ahead and close it.

I currently have a card header component that i use like so:

```
{% component "card_header" ...other code... tooltip_text=_("Responsible for the organisation's data protection compliance") / %}
```

If the tag is not closing via the `/ %}` syntax like so:

```
{% component "card_header" ...other code... tooltip_text=_("Responsible for the organisation's data protection compliance") %} {% endcomponent %}
```

Then i actually get the following error:
IndexError: pop from empty list

I have narrowed it down to the fact that i have an apostrophe ( ' ) in my text and causes and issue when the template is being parsed. That means i do NOT get the error if i:

1. Remove the apostrophe from my str variable.
2. Let the tag be self-closing.

Is this a known issue?

**Comment by EmilStenstrom** [comment]
Hi Martin, it sounds like a bug that should be fixed. 

@JuroOravec Mind helping me understand what the right syntax should be here? `tooltip_text="{% trans "Responsible..." %}"`?

**Comment by JuroOravec** [comment]
@MartinBoge Do you have access to the full stack trace? Could you post it here?

Martin's example SHOULD work, so this is indeed a bug.

Btw, @MartinBoge, so if you omit the single apostrophe, then you are able to pass the text like so?

```django
{% component "card_header" tooltip_text=_("Responsible for the organisations data protection compliance") / %}
```

I haven't worked with Django's translation / internationalisation features, so personally I never tried passing the `key=_("some text")` syntax through components. I wonder if that has something to do with it?

@MartinBoge could you check which of the following syntaxes work correctly? The one you mentioned above is case 7. below. I expect that cases 1-4 SHOULD work, but no idea how 5-8 behave.

1. Without `_()`, without single apostrophe, self-closing:
	```django
	{% component "card_header" tooltip_text="organisations" / %}
	```

2. Without `_()`, without single apostrophe, end-tag:
	```django
	{% component "card_header" tooltip_text="organisations" %} {% endcomponent %}
	```

3. Without `_()`, with single apostrophe, self-closing:
	```django
	{% component "card_header" tooltip_text="organisation's" / %}
	```

4. Without `_()`, with single apostrophe, end-tag:
	```django
	{% component "card_header" tooltip_text="organisation's" %} {% endcomponent %}
	```

5. With `_()`, without single apostrophe, self-closing:
	```django
	{% component "card_header" tooltip_text=_("organisations") / %}
	```

6. With `_()`, without single apostrophe, end-tag:
	```django
	{% component "card_header" tooltip_text=_("organisations") %} {% endcomponent %}
	```

7. With `_()`, with single apostrophe, self-closing:
	```django
	{% component "card_header" tooltip_text=_("organisation's") / %}
	```

8. With `_()`, with single apostrophe, end-tag:
	```django
	{% component "card_header" tooltip_text=_("organisation's") %} {% endcomponent %}
	```

I'll be able to have a look at this towards the end of this week  :)

**Comment by JuroOravec** [comment]
I was able to pinpoint this to the cases 4. and 8., so the issue is with the combination of a single quote and the endtag.

```django
{% component "test" text="organisation's" %} {% endcomponent %}
{% component "test" text=_("organisation's") %} {% endcomponent %}
```

It relates to the logic we use to allow nesting template tags inside tag args / kwargs (See https://github.com/JuroOravec/django-components/blob/12b7c6c0a8da9db723f9e80f070ed2bc61e0f7da/src/django_components/templatetags/component_tags.py#L1065).

Currently we do a quick and dirty trick that depends on the number of apostrophes in the `{% component %}` tag.

E.g. if we consider the example we've used above:

```django
{% component "card_header" tooltip_text=_("organisation's") %} {% endcomponent %}
```

then the `component "card_header"` is removed internally, and that leaves us with 

```txt
tooltip_text=_("organisation's")
```

Which has 2 double quotes and 1 single quote. It's the odd number of single quotes which triggers the wrong logic.

---

The fix for this would be to properly parse the content within `{% component %}` tag, so we ignore quotes that are part of a string.

I had already written such parser / tokenizer when working towards django-vue. So I'll just copy the code and adapt it.

**Comment by MartinBoge** [comment]
Sorry for getting back to you late @JuroOravec.

I can confirm you have pinpointed the bug correctly. Is there anything i can do to assist further?

**Comment by JuroOravec** [comment]
@MartinBoge No, not with this one :) I've got the fix written up, and we'd release it as part of 0.110

**Comment by JuroOravec** [comment]
@MartinBoge The fix has now been released in [v0.110](https://github.com/EmilStenstrom/django-components/releases/tag/0.110) :)

**Comment by MartinBoge** [comment]
Thank you very much!
